src/plots/attributes.js

```diff
@@ -41,6 +41,16 @@ module.exports = {
             'when toggling legend items.'
         ].join(' ')
     },
+    legendrank: {
+        valType: 'number',
+        dflt: 1000,
+        editType: 'style',
+        description: [
+            'Sets the legend rank for this trace.',
+            'Items with smaller ranks would be presented on top/left side while',
+            'with `*reversed* `legend.traceorder` they would be on bottom/right side.'
```

Comment:
Let's describe what the `dflt: 1000` is for, something like:
> The default `legendrank` is 1000, so that you can use ranks less than 1000 to place certain items before all unranked items, and ranks greater than 1000 to go after all unranked items.

Reply:
Thanks. Please see https://github.com/plotly/plotly.js/pull/5591/commits/4ffb06c5b283b72444ab3613e4bb8b573bfac057 I also added a note to mention that ranking has impact within `legendgroup`.